port/unix/omrsysinfo.c

```diff
@@ -1564,14 +1585,25 @@ omrsysinfo_get_s390_description(struct OMRPortLibrary *portLibrary, OMRProcessor
 	/* zEC12 facility and processor detection */
 
 	/* TE/TX hardware support */
-	if (omrsysinfo_test_stfle(portLibrary, 50) && omrsysinfo_test_stfle(portLibrary, 73)) {
+	if (omrsysinfo_test_stfle(portLibrary, 73)) {
 #if defined(J9ZOS390)
 		if (omrsysinfo_get_s390_zos_supports_transactional_execution_facility())
 #elif defined(LINUX) /* LINUX S390 */
 		if (OMR_ARE_ALL_BITS_SET(auxvFeatures, OMR_HWCAP_S390_TE))
 #endif /* defined(J9ZOS390) */
 		{
-			omrsysinfo_set_feature(desc, OMR_FEATURE_S390_TE);
+			omrsysinfo_set_feature(desc, OMR_FEATURE_S390_TRANSACTIONAL_EXECUTION_FACILITY);
+		}
+	}
+
+	if (omrsysinfo_test_stfle(portLibrary, 50)) {
+#if defined(J9ZOS390)
+		if (omrsysinfo_get_s390_zos_supports_constrained_transactional_execution_facility())
+#elif defined(LINUX) /* LINUX S390 */
+		if (OMR_ARE_ALL_BITS_SET(auxvFeatures, OMR_HWCAP_S390_TE))
```

Comment:
> Should we update auxv features as well though to keep it more current ?

@r30shah : Yeah, that's a good point.  The various vector extensions and deflate hwcap bits won't require additional OS support (beyond base vector support).  We will need to confirm under which kernel levels these HWCAPs will be updated.  Otherwise, we risk assuming that the features are unavailable if the respective hwcap bits are not set.  As such, while I think we can update the list in [2] to be more comprehensive, I don't think we should add these extra hwcap tests, unless our min kernel levels are guaranteed to properly set the tested hwcap bits.

Reply:
Thanks @joransiu , Opened up the issue https://github.com/eclipse/omr/issues/7121 to keep track of that. 

Irrespective of what we do with that, Can I request your review as well on this PR ?